From 26b49efbb6150bdd148f5c68f66eee7fe4f4fb2e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 1 May 2022 15:15:19 +0200
Subject: [PATCH] Set SWIGWORDSIZE dependent on architecture bitness

Setting SWIGWORDSIZE to 64bit on 32bit archs breaks build on
32 bit archs, e.g. i586 or armv7.

See #3271.
---
 cmake/python.cmake | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/cmake/python.cmake b/cmake/python.cmake
index c29eddd2d0..0568b22383 100644
--- a/cmake/python.cmake
+++ b/cmake/python.cmake
@@ -19,7 +19,11 @@ if(${SWIG_VERSION} VERSION_GREATER_EQUAL 4)
 endif()
 
 if(UNIX AND NOT APPLE)
-  list(APPEND CMAKE_SWIG_FLAGS "-DSWIGWORDSIZE64")
+  if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+    list(APPEND CMAKE_SWIG_FLAGS "-DSWIGWORDSIZE64")
+  else()
+    list(APPEND CMAKE_SWIG_FLAGS "-DSWIGWORDSIZE32")
+  endif()
 endif()
 
 # Find Python 3
-- 
2.36.1

